### ISSUE TITLE
Markup code sections for screen readers

### ISSUE THREAD

**Issue by janosdebugs** [opentofu/opentofu.org#260]
Currently, the code blocks are not marked up to make them easy to navigate for screen readers. This causes them to be read in line with the rest of the text.

There are several solutions to make them stand out. The easiest is to wrap them in a `<figure></figure>`. More advanced options could be adding a [`role` attribute](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles) to the block. A `<figcaption></figcaption>` can also be used to enhance the experience alongside the appropriate CSS rules:

```html
<figure aria-roledescription="Code block">
    <!- This needs automation to inject the language: -->
    <figcaption class="sr-only">Language: hcl</figcaption>
    <pre>...</pre>
</figure>
```

**Comment by SypherSP** [comment]
Hi @janosdebugs, i would like to have a go at this issue. 

Could you please help me out with the relevant part of the code that is responsible for rendering the .mdx files?

**Comment by janosdebugs** [comment]
Hey @SypherSP sorry I missed this, I've assigned you. Regarding the specific implementation, in order to make the code more accessible, we should do the following (in order, happy to have separate PRs for these):

1. Make sure that the code block has a `tabindex="0"` defined so that the screen reader can focus on it with keyboard navigation.
2. The code could be wrapped in a figure to provide additional context using the figcaption. The screen reader-only functions should have a CSS markup to make it visible to screen readers and Braille users, but not to "normal" browsers.

To test this feature you can:

1. Use the free NVDA software.
2. Use the Android Talkback or iOS Voiceover feature.

All 3 of these require some getting used to, please familiarize yourself with them so you can test this properly. Specifically, please look at the browse vs. focus mode when using NVDA, these can be a bit confusing at first.

**Comment by SypherSP** [comment]
> ```
> <!- This needs automation to inject the language: -->
> ```
Could you explain what you mean by this. 

What do you think about implementing a new component like 
```
const FigureWithCaption: React.FC<FigureWithCaptionProps> = ({ children, caption }) => {
  return (
    <figure aria-roledescription="Code block">
    <figcaption className="sr-only" tabIndex={0}>Caption: {caption}</figcaption>
    {children}
</figure>
  );
};
```
which can be used in any mdx file like this 
```
<FigureWithCaption caption="Sample code">
``hcl
terraform {
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "~> 1.0.4"
    }
  }
}
``
</FigureWithCaption>
```

**Comment by SypherSP** [comment]
Okay, ill look for another approach

**Comment by janosdebugs** [comment]
@Yantrio it may not be possible, we may have to create a subclass(?) of `<CodeBlock>` to improve this, but I agree that upstreaming the fixes would be preferable.

**Comment by SypherSP** [comment]
Using Swizzling we can add the figure wrapper easily, but to add a caption from markdown might be a little complicated.

**Comment by SypherSP** [comment]
Yeah, im testing things as im adding them. Will link a pr as soon as i figure out wrapping.

**Comment by codiini** [comment]
@janosdebugs I think this can be closed since the corresponding PR has been merged
